main.js

```diff
@@ -1,32 +1,192 @@
+        [new Block(-1, 0), new Block(0, 0), new Block(0, -1), new Block(1, 0)],
+        [new Block(0, 0), new Block(1, 0), new Block(0, -1), new Block(0, -2)],
+      ]
+      return blocks[this.tetroType].map((block) => {
+        return new Block(this.x + block.x, this.y + block.y)
+      })
+    }
+  }
+
+  class Field {
+    constructor() {
+      this.tiles = [
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1],
+        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+      ]
+    }
+    draw() {
+      for (let row = 0; row < field_vertical; row++) {
+        for (let col = 0; col < field_beside; col++) {
+          if (this.tiles[row][col]) {
+            drawBlock(col, row)
+          }
+        }
+      }
+    }
+    tileAt(x, y) {
+      if ((x < 0 && x > field_beside) || (y < 0 && y > field_vertical)) {
+        return 1
+      }
+      return this.tiles[y][x]
+    }
+    putBlock(x, y) {
+      //下方向に移動できないとき、fieldの座標を埋める
+      this.tiles[y][x] = 1
+    }
+  }
+
+  class Board {
+    constructor() {
+      this.tetroMino = new Tetromino(5, 3)
+      this.intervalID
+      this.gameOver = false
+    }
+    init() {
+      this.field = new Field()
+      this.field.draw()
+      this.tetroMino.draw()
+      this.moveTetroMino()
+      this.gameOver = false
+      this.intervalID = setInterval(() => {
+        const m = this.tetroMino.copy()
+        m.y++
+        conText.clearRect(0, 0, canvas_beside, canvas_vertical)
+        this.field.draw()
+        if (this.checkMove(m, this.field) == true) {
+          this.tetroMino.y++
+        } else {
+          this.tetroMino.getBlocks().forEach((block) => {
+            this.field.putBlock(block.x, block.y)
+            this.field.draw()
+          })
+          this.tetroMino = new Tetromino(6, 3)
+          this.tetroMino.draw()
+          if (this.checkMove(this.tetroMino, this.field) == false) {
+            this.gameOver = true
+            clearInterval(this.intervalID)
+            confirm("Game Over")
+          }
+        }
+        this.tetroMino.draw()
+      }, 1000)
+    }
+    checkMove(copy, field) {
```

Comment:
```suggestion
    checkMove(tetro, field) {
```

変数名はオブジェクトのことがわかるようにしたい